src/core/utils.js

```diff
@@ -494,6 +495,25 @@ export function flatten(collector, item) {
 
 // --- DOM HELPERS -------------------------------
 
+// Takes an array and returns a string that separates each of its items with the proper commas and
+// "and". The second argument is a mapping function that can convert the items before they are
+// joined
+// Finally converts to hyperHTML while joining them
+// htmlJoinAnd(["x"])).toEqual(hyperHTML`x`);
+export function htmlJoinAnd(array = [], mapper = item => hyperHTML`${item}`) {
```

Comment:
I don't think array can be undefined. If it is, it's a usage error.
Maybe use JSDoc/ts-check to ensure that it's an error.

Reply:
TS strict mode is needed to check `undefined`. Not sure we are ready to be strict though 😥

Reply:
159 errors with `strictNullChecks` :grin: 
I think it's fine to add JSDoc here enforcing `any[]`. It's cleaner in terms of this util API (and we're the sole users of this util)

Reply:
Final nit (sorry to add again and again 😭):

```suggestion
export function htmlJoinAnd(array, mapper = item => item) {
```

Removing array default value as @sidvishnoi mentioned, and reverting mapper default value because `` hyperHTML`${item}` `` doesn't really do anything here (it's near no-op).